### PR TITLE
fs: Fix checking for UID/GID == 0

### PR DESCRIPTION
**Pull request by vojtechtrefny**
We need to check if the value was "0", not option itself.

### CODE REVIEW

File: src/plugins/fs/mount.c [hunk 1]
```diff
@@ -618,7 +618,7 @@ gboolean bd_fs_unmount (const gchar *spec, gboolean lazy, gboolean force, const
                 run_as_uid = g_ascii_strtoull ((*extra_p)->val, NULL, 0);
 
                 /* g_ascii_strtoull returns 0 in case of error */
-                if (run_as_uid == 0 && (g_strcmp0 ((*extra_p)->opt, "0") != 0)) {
+                if (run_as_uid == 0 && (g_strcmp0 ((*extra_p)->val, "0") != 0)) {
```
Comment (tbzatek) [inline src/plugins/fs/mount.c]: Good catch, however...

This is just a weird test. Conversion error detection should be done by checking the value returned via the second argument of `g_ascii_strtoull()`.

Comment (vojtechtrefny) [reply src/plugins/fs/mount.c]: Ok, time to do that correctly.

File: src/plugins/fs/mount.c [hunk 2]
```diff
@@ -615,19 +616,19 @@ gboolean bd_fs_unmount (const gchar *spec, gboolean lazy, gboolean force, const
     if (extra) {
         for (extra_p=extra; *extra_p; extra_p++) {
             if ((*extra_p)->opt && (g_strcmp0 ((*extra_p)->opt, "run_as_uid") == 0)) {
-                run_as_uid = g_ascii_strtoull ((*extra_p)->val, NULL, 0);
+                run_as_uid = g_ascii_strtoull ((*extra_p)->val, endptr, 0);
 
                 /* g_ascii_strtoull returns 0 in case of error */
-                if (run_as_uid == 0 && (g_strcmp0 ((*extra_p)->opt, "0") != 0)) {
+                if (run_as_uid == 0 && endptr == &((*extra_p)->val)) {
```
Comment (vpodzime) [inline src/plugins/fs/mount.c]: Yeah, this is so much nicer :smile:

File: src/plugins/fs/mount.c [hunk 3]
```diff
@@ -689,19 +691,19 @@ gboolean bd_fs_mount (const gchar *device, const gchar *mountpoint, const gchar
     if (extra) {
         for (extra_p=extra; *extra_p; extra_p++) {
             if ((*extra_p)->opt && (g_strcmp0 ((*extra_p)->opt, "run_as_uid") == 0)) {
-                run_as_uid = g_ascii_strtoull ((*extra_p)->val, NULL, 0);
+                run_as_uid = g_ascii_strtoull ((*extra_p)->val, endptr, 0);
```
Comment (vpodzime) [inline src/plugins/fs/mount.c]: Should be `&endptr` with `gchar *endptr`, AFAICT.

Comment (vojtechtrefny) [reply src/plugins/fs/mount.c]: Oops, fixed.

**Pull request merged**